tests/test_buffer.py

```diff
@@ -137,6 +137,19 @@ def test_SimpleArray(self):
         self.assertEqual((1, 24), sarr.reshape((1, 24)).shape)
         self.assertEqual((12, 2), sarr.reshape((12, 2)).shape)
 
+    def test_SimpleArray_invalid_ghost(self):
+        sarr = modmesh.SimpleArrayInt8(10)
+        with self.assertRaisesRegex(
+            IndexError, r"SimpleArray: cannot set nghost 11 > shape\(0\) 10"
+        ):
+            sarr.nghost = 11
```

Comment:
The tests are useful.  Thanks for adding them.

It would be even better to put them in `test_SimpleArray_ghost_1d()` test function to be close with other ghost index testing.  The two assertions here are far away from the example code that set `nghost` in the right way.